contracts/strategies/RFPSimple.sol

```diff
@@ -0,0 +1,185 @@
+
+    function reviewRecipients(ApplicationStatus[] memory statuses, uint256 refRecipientsCounter)
+        public
+        override
+        onlyActiveRegistration
+    {
+        bool recipientAccepted;
+        for (uint256 i; i < statuses.length;) {
+            uint256 fullRow = statuses[i].statusRow;
+            for (uint256 col = 0; col < 64;) {
+                uint8 status = uint8((fullRow >> col * 4) & 0xF);
+                if (IRecipientsExtension.Status(status) == IRecipientsExtension.Status.Accepted) {
+                    if (recipientAccepted) {
+                        // Only one recipient can be accepted
+                        revert INVALID();
+                    } else {
+                        recipientAccepted = true;
+                        registrationEndTime = uint64(block.timestamp - 1);
+                    }
+                }
+                unchecked {
+                    col++;
+                }
+            }
+            unchecked {
+                i++;
+            }
+        }
+        super.reviewRecipients(statuses, refRecipientsCounter);
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Hook to process recipient data
+    /// @param _recipientId ID of the recipient
+    /// @param _extraData The extra data of the recipient
+    function _processRecipient(address _recipientId, bool, Metadata memory, bytes memory _extraData)
+        internal
+        virtual
+        override
+    {
+        uint256 proposalBid = abi.decode(_extraData, (uint256));
+        _setProposalBid(_recipientId, proposalBid);
+    }
+
+    /// @notice This will allocate to recipients.
+    function _allocate(address[] memory, uint256[] memory, bytes memory, address) internal virtual override {
+        revert NOT_IMPLEMENTED();
+    }
+
+    /// @notice Distributes funds (tokens) to recipients.
+    /// @dev The encoded '_data' will be determined by the strategy implementation. Only 'Allo' contract can
+    ///      call this when it is initialized.
+    /// @param _recipientIds The IDs of the recipients
+    /// @param _data The data to use to distribute to the recipients
+    /// @param _sender The address of the sender
+    function _distribute(address[] memory _recipientIds, bytes memory _data, address _sender)
+        internal
+        virtual
+        override
+    {
+        if (block.timestamp <= registrationEndTime) {
+            revert ALLOCATION_NOT_ENDED();
+        }
+        if (_recipientIds.length > 1) {
+            revert RECIPIENT_NOT_ACCEPTED();
+        }
+
+        address acceptedRecipientId = _recipientIds[0];
+        uint256[] memory milestoneIds = abi.decode(_data, (uint256[]));
+        uint256 amount;
+        for (uint256 i = 0; i < milestoneIds.length; i++) {
+            uint256 milestoneId = milestoneIds[i];
+            Milestone storage milestone = milestones[milestoneId];
```

Comment:
I think you can use `memory` here.

Reply:
No because only `milestone.status` is used. Using memory would copy from storage to memory the entire struct. Just pushed a change to use `milestones[milestoneId].status` directly though.

Reply:
Interesting